bullet_train-fields/app/controllers/concerns/fields/super_select_support.rb

```diff
@@ -20,10 +24,35 @@ def create_model_if_new(id)
   end
 
   def create_models_if_new(ids)
+    ActiveSupport::Deprecation.warn(
+      "#create_models_if_new is deprecated. " \
+      "Use #ensure_valid_ids_or_create_model instead. See an example at https://bullettrain.co/docs/field-partials/super-select#accepting-new-entries"
+    )
     ids.map do |id|
       create_model_if_new(id) do
         yield(id)
       end
     end
   end
+
+  def ensure_valid_id_or_create_model(id_or_string, collection: [], attribute: :id)
```

Comment:
We can't add an Array default attribute for `collection` since we're using Active Record API on it. I think it should be required too, otherwise this method isn't validating anything. Same goes for the plural version.

```suggestion
  def ensure_valid_id_or_create_model(id_or_string, collection:, attribute: :id)
```